README.md

```diff
@@ -106,8 +106,36 @@ The following are available:
   `{"url":..., "allow":...}` where `url` is the Shesmu server to copy and
   `allow` is a regular expression of which actions to copy.
 
-### Variable Sources
-This is where Shesmu gets provenance data for olives to ingest. The following are available:
+### Source Definitions
+A _source_ is a format of data that Shesmu can process. This describes the
+variables that are available to Shesmu programs. The actual data comes from a
+matching _source repository_ and many repositories can provide the same format.
+
+- `gsi_std` is the default format. It contains a mix of analysis provenance
+  information tied back to LIMS provenance data.
+
+Creating a new source format is meant to be easy, but convoluted in order to be
+type safe. To create a new source format:
+
+1. Create a class, _I_, that will hold all the data for a “row” in the incoming
+data. It must be a class and not an interface.
+1. Create a parameterless method in _I_ for every variable to be exposed. The
+method names must be valid Shemsu names (lowercase with underscores) and
+decorated with `@Export` annotations with the correct type signature. All
+methods must return `boolean`, `long`, `String`, `Instant`, `Set`, or `Tuple`
+(and `Set` and `Tuple` may only contain more of the same).
+1. Create a new interface, _R_, extending `SourceRepository` that fills in the type
```

Comment:
I wonder if it makes sense to use `_I_` for the interface? It'll make it slightly easier to keep track of which is the class and which is the interface below.